gateway/api/views.py

```diff
@@ -86,11 +87,10 @@ def run(self, request):
         tracer = trace.get_tracer("gateway.tracer")
         ctx = TraceContextTextMapPropagator().extract(carrier=request.headers)
         with tracer.start_as_current_span("gateway.program.run", context=ctx):
-            serializer = self.get_serializer(data=request.data)
```

Comment:
serializer should handle all related entities, so you do not need to created them.

Reply:
I restore the use of the ProgramSerializer.  Thanks!